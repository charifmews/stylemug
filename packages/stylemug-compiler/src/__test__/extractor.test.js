const extractor = require('../extractor');

describe('extractor', () => {
  it('should generate basic css', () => {
    extractor.save('a', {
      key: 'color',
      value: 'red',
      children: '',
      media: null,
    });

    extractor.save('b', {
      key: 'color',
      value: 'blue',
      children: ':hover',
      media: null,
    });

    expect(extractor.flushAsCss()).toMatchSnapshot();
  });

  it('should hyphenate keys', () => {
    const rule = extractor.createRule('a', {
      key: 'fontSize',
      value: 16,
      children: '',
      media: null,
    });
    expect(rule).toBe('.a{font-size:16}');
  });
});
