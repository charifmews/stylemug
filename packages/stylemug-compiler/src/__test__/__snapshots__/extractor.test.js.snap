// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractor should generate basic css 1`] = `".a{color:red}.b:hover{color:blue}"`;
